Source/Icebreaker/Bot/IcebreakerBot.cs

```diff
@@ -314,6 +409,138 @@ private async Task HandleMessageActivityAsync(ITurnContext turnContext, Cancella
             }
         }
 
+        /// <summary>
+        /// Send view teams card
+        /// </summary>
+        /// <param name="turnContext">Context object containing information cached for a single turn of conversation with a user.</param>
+        /// <param name="userInfo">User info</param>
+        /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private async Task SendViewTeamsCardAsync(ITurnContext turnContext, UserInfo userInfo, CancellationToken cancellationToken)
+        {
+            var teamNameLookup = await this.GetTeamNamesAsync(userInfo, turnContext.Adapter);
+            var teamsViewCard = MessageFactory.Attachment(TeamsViewCard.GetTeamsViewCard(userInfo, teamNameLookup));
+            var response = await turnContext.SendActivityAsync(teamsViewCard, cancellationToken);
+            this.teamsViewCardId = response.Id;
+        }
+
+        /// <summary>
+        /// Handle adaptive card submits
+        /// </summary>
+        /// <param name="activity">Message from submitted card</param>
+        /// <param name="turnContext">Context object containing information cached for a single turn of conversation with a user.</param>
+        /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private async Task OnAdaptiveCardSubmitAsync(Activity activity, ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var cardPayload = JToken.Parse(activity.Value.ToString());
+            var cardType = cardPayload["ActionType"].Value<string>().ToLowerInvariant();
+            var teamId = activity.Conversation.Id;
+            var userId = activity.From.Id;
+            var userInfo = await this.dataProvider.GetUserInfoAsync(userId);
+
+            // CLEAN UP HARDCODED TEXT
+            switch (cardType)
+            {
+                // updated pause preferences
+                case "saveopt":
+
+                    // update database
+                    this.telemetryClient.TrackTrace("Received pause preferences");
+                    var optedIn = new Dictionary<string, bool>();
+                    var activeTeams = new Dictionary<string, string>();
+                    var botAdapter = turnContext.Adapter;
+
+                    foreach (var team in userInfo.OptedIn.Keys.ToList())
+                    {
+                        if (cardPayload[team] != null)
```

Comment:
NIT: I would either filter out the `OptedIn.Keys` so that it is only those where `cardPayload[team] != null` or invert this condition & use a `continue` statement to avoid the extra nesting.